benchmarks/drop/drop.py

```diff
@@ -0,0 +1,381 @@
+"""DROP: A Reading Comprehension Benchmark Requiring Discrete Reasoning Over Paragraphs
+
+Dheeru Dua, Yizhong Wang, Pradeep Dasigi,
+Gabriel Stanovsky, Sameer Singh, and Matt Gardner
+https://arxiv.org/pdf/1903.00161
+
+Based on: https://github.com/openai/simple-evals/blob/main/drop_eval.py
+and https://github.com/EleutherAI/lm-evaluation-harness/tree/main/lm_eval/tasks/drop
+
+# run with default fewshot samples (3)
+inspect eval drop/drop.py
+
+# run with more or no fewshot samples
+inspect eval drop/drop.py -T fewshot=5
+inspect eval drop/drop.py -T fewshot=false
+"""
+
+import re
+import string
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+from scipy.optimize import linear_sum_assignment
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import Sample, hf_dataset
+from inspect_ai.scorer import Score, Target, bootstrap_std, mean, scorer
+from inspect_ai.solver import (
+    Generate,
+    TaskState,
+    generate,
+    prompt_template,
+    solver,
+    system_message,
+)
+
+DROP_PROMPT_TEMPLATE = {
+    # Based on the prompt provided here: https://github.com/openai/simple-evals/blob/main/drop_eval.py#L261C13-L283C91
+    "system": """You will be asked to read a passage and answer a question.\n\n""",
+    "system_w_examples": """You will be asked to read a passage and answer a question. Some examples of passages and Q&A are provided below.\n\n"""
+    + """Examples\n{examples}\n\n""",
+    "user": """Your Task\n---\n{prompt}\n\n"""
+    + """Think step by step, then write a line of the form "Answer: $ANSWER" at the end of your response.\n""",
+}
+
+# Borrowed from: https://github.com/openai/simple-evals/blob/main/common.py#L24
+ANSWER_PATTERN = r"(?i)Answer\s*:\s*([^\n]+)"
+
+
+@task
+def drop(
+    fewshot: int = 3,
+    fewshot_seed: int = 42,
+) -> Task:
+    """Inspect task implementing the DROP benchmark.
+
+    Arguments:
+        fewshot (int): Number of few shot examples to use.
+        fewshot_seed (int): Random seed for sampling few shot examples.
+    """
+    # Get prompt
+    prompt_dict = DROP_PROMPT_TEMPLATE
+    prompt_user = prompt_dict["user"]
+    prompt_sys = prompt_dict["system"]
+    prompt_sys_w_examples = prompt_dict["system_w_examples"]
+
+    # Add system message based on zero-shot or few-shot prompting
+    if fewshot:
```

Comment:
Maybe a function that returns the system message (and takes the few shot args?)

Reply:
Created a separate function to build the plan incorporating the zero-shot and few-shot system messages.